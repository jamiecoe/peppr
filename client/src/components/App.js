import React, { Component } from 'react';
import { Router, Route, Redirect, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { getUser } from '../actions/auth';

import createLoadable from './helpers/createLoadable';
import history from '../actions/history';

const Landing = createLoadable({ loader: () => import('./auth/Landing') });
const AddRecipe = createLoadable({ loader: () => import('./addrecipe/AddRecipe') });
const Recipes = createLoadable({ loader: () => import('./recipes/Recipes') });
const SingleRecipe = createLoadable({ loader: () => import('./recipes/SingleRecipe') });
const ServerError = createLoadable({ loader: () => import('./errors/ServerError500') });
const NotFound = createLoadable({ loader: () => import('./errors/NotFound400') });

export class App extends Component {
  componentDidMount() {
    this.props.getUser();
  }

  render() {
    if (this.props.auth === null) return <div />;
    return (
      <Router history={history}>
        <Switch>
          <Route
            exact
            path="/"
            render={props =>
              (!this.props.auth ? (
                <Landing {...props} />
              ) : (
                <Redirect to="/recipes" />
              ))
            }
          />

          <Route
            exact
            path="/signup"
            render={props =>
              (!this.props.auth ? (
                <Landing {...props} />
              ) : (
                <Redirect to="/recipes" />
              ))
            }
          />

          <Route
            exact
            path="/recipes"
            render={() => (this.props.auth ? <Recipes /> : <Redirect to="/" />)}
          />

          <Route
            exact
            path="/recipes/:id"
            render={props =>
              (this.props.auth ? (
                <SingleRecipe {...props} />
              ) : (
                <Redirect to="/" />
              ))
            }
          />

          <Route
            exact
            path="/addrecipe"
            render={() =>
              (this.props.auth ? <AddRecipe /> : <Redirect to="/" />)
            }
          />

          <Route
            exact
            path="/servererror"
            render={() =>
              (this.props.auth ? <ServerError /> : <Redirect to="/" />)
            }
          />

          <Route component={NotFound} />
        </Switch>
      </Router>
    );
  }
}

const mapStateToProps = state => ({ auth: state.auth });

export default connect(mapStateToProps, { getUser })(App);
