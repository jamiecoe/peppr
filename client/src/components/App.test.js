import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('App', () => {
  const mockGetUser = jest.fn();

  const props = {
    getUser: mockGetUser,
    auth: true,
  };

  let app = shallow(<App {...props} />, {
    disableLifecycleMethods: false,
  });

  it('should render properly, as an authenticated user', () => {
    expect(app).toMatchSnapshot();
  });

  it('should call getUser() in componentDidMount', () => {
    expect(mockGetUser).toHaveBeenCalled();
  });

  describe('when the user is not authenticated', () => {
    beforeEach(() => {
      props.auth = false;
      app = shallow(<App {...props} />);
    });

    it('should render properly, as an unauthenicated user', () => {
      expect(app).toMatchSnapshot();
    });
  });

  describe('when the getUser() has not returned, and props.auth is null', () => {
    beforeEach(() => {
      props.auth = null;
      app = shallow(<App {...props} />);
    });

    it('should render properly', () => {
      expect(app).toMatchSnapshot();
    });
  });
});
