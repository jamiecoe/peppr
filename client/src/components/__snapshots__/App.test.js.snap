// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly, as an authenticated user 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    auth={true}
    getUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Switch>
        <Route
                exact={true}
                path="/"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/signup"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/recipes"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/recipes/:id"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/addrecipe"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/servererror"
                render={[Function]}
        />
        <Route
                component={[Function]}
        />
</Switch>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/signup"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/recipes"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/recipes/:id"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/addrecipe"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/servererror"
            render={[Function]}
/>,
          <Route
            component={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/signup",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/recipes",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/recipes/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/addrecipe",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/servererror",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/signup"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/recipes"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/recipes/:id"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/addrecipe"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/servererror"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
          />
</Switch>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/signup"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/recipes"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/recipes/:id"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/addrecipe"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/servererror"
              render={[Function]}
/>,
            <Route
              component={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/signup",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/recipes",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/recipes/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/addrecipe",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/servererror",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": false,
  },
}
`;

exports[`App when the getUser() has not returned, and props.auth is null should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    auth={null}
    getUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`App when the user is not authenticated should render properly, as an unauthenicated user 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    auth={false}
    getUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Switch>
        <Route
                exact={true}
                path="/"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/signup"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/recipes"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/recipes/:id"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/addrecipe"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/servererror"
                render={[Function]}
        />
        <Route
                component={[Function]}
        />
</Switch>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/signup"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/recipes"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/recipes/:id"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/addrecipe"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/servererror"
            render={[Function]}
/>,
          <Route
            component={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/signup",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/recipes",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/recipes/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/addrecipe",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/servererror",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/signup"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/recipes"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/recipes/:id"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/addrecipe"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/servererror"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
          />
</Switch>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/signup"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/recipes"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/recipes/:id"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/addrecipe"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/servererror"
              render={[Function]}
/>,
            <Route
              component={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/signup",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/recipes",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/recipes/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/addrecipe",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/servererror",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
